File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/link/LinkDao.java
package ru.nikshlykov.englishwordsapp.db.link;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface LinkDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    long insert(Link link);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    List<Long> insertMultiple(List<Link> links);

    @Delete
    int delete(Link link);

    @Delete
    int deleteMultiple(List<Link> links);

    @Query("SELECT * FROM Links WHERE WordId = :wordId")
    Link[] getLinksByWordId(long wordId);

    @Query("SELECT * FROM Links WHERE WordId = :wordId AND SubgroupId = :subgroupId")
    Link getLink(long wordId, long subgroupId);
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/link/LinkDao.kt
package ru.nikshlykov.englishwordsapp.db.link

import androidx.room.*

@Dao
interface LinkDao {

  @Insert(onConflict = OnConflictStrategy.IGNORE)
  suspend fun insert(link: Link): Long

  @Delete
  suspend fun delete(link: Link): Int

  @Delete
  fun deleteMultiple(links: List<Link>): Int

  @Query("SELECT * FROM Links WHERE WordId = :wordId")
  suspend fun getLinksByWordId(wordId: Long): List<Link>

  @Query("SELECT * FROM Links WHERE WordId = :wordId AND SubgroupId = :subgroupId")
  fun getLink(wordId: Long, subgroupId: Long): Link
}